internal/service/tesla.go

```diff
@@ -0,0 +1,129 @@
+package service
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"github.com/DIMO-Network/shared/pkg/cipher"
+	"github.com/DIMO-Network/shared/pkg/db"
+	"github.com/DIMO-Network/tesla-oracle/internal/config"
+	dbmodels "github.com/DIMO-Network/tesla-oracle/models"
+	"github.com/rs/zerolog"
+	"regexp"
+	"strconv"
+)
+
+type TeslaService struct {
+	settings *config.Settings
+	logger   *zerolog.Logger
+	Cipher   cipher.Cipher
+	pdb      *db.Store
+}
+
+func NewTeslaService(settings *config.Settings, logger *zerolog.Logger, cipher cipher.Cipher, pdb *db.Store) *TeslaService {
+	return &TeslaService{
+		settings: settings,
```

Comment:
suggestion: One thing I have started doing is instead of storing the setting's field in the struct I pull off what fields I am actually using and set them in new.

This make it easier for someone to look at the TeslaService and know what settings it needs when updating the code, or testing since they don't have to find all the usages.